### PR TITLE
Add cask: Visual Studio for Mac preview-1

### LINKED ISSUE #26882: Cask request: visual-studio-mac
### Cask details

(Please fill out as much as possible)

**Name**:visual-studio-mac

**Homepage**: https://www.visualstudio.com/vs/visual-studio-mac/

**Download URL**:https://www.visualstudio.com/thank-downloading-visual-studio-mac/?sku=vsmac

**Description**:A mobile-first, cloud-first IDE. Made for the Mac.

### PR DESCRIPTION
**Pull request by joshka**
- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] The commit message includes the cask’s name and version.
- [x] Named the cask according to the [token reference].
- [x] `brew cask install {{cask_file}}` worked successfully.
- [x] `brew cask uninstall {{cask_file}}` worked successfully.
- [x] Checked there are no [open pull requests] for the same cask.
- [x] Checked that the cask was not already refused in [closed issues].

Note, this is distinct from Visual Studio Code. They can be installed
concurrently. I was unable to find any info on the installer arguments
for unattended installation, so this will show the user a UI.

See https://www.visualstudio.com/vs/visual-studio-mac/ to confirm download link.
Fixes #26882

### CODE REVIEW

**Comment by jawshooah** [comment]
> I was unable to find any info on the installer arguments for unattended installation, so this will show the user a UI.

In that case, this needs `installer :manual`, not `:script`.

**Comment by joshka** [comment]
> In that case, this needs installer :manual, not :script.
Done

**Comment by jawshooah** [comment]
Looks like it's still using `script`. Are you sure you pushed the right branch?

**Comment by joshka** [comment]
Heh, nope ;) Fixed it now.

Is changing to manual all that's required there, or does the path change as well?

File: Casks/visual-studio.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  sha256 '7c674f7114e6a6532d6498dd8963cf1b5916badd694f251a11edf3024809bf48'
+
+  # xamarin.azureedge.net was verified as official when first introduced to the cask
+  url "https://xamarin.azureedge.net/VsMacInstaller/#{version.after_comma}/VisualStudioforMacPreviewInstaller.dmg"
+  name 'Visual Studio for Mac'
+  homepage 'https://www.visualstudio.com/vs/visual-studio-mac/'
+
+  installer manual: 'Install Visual Studio for Mac Preview.app/Contents/MacOS/Install_Xamarin'
+
+  uninstall delete: '/Applications/Visual Studio.app'
```
Comment (jawshooah) [inline Casks/visual-studio.rb]: Seems unlikely that the installer just unpacks an app bundle and drops it in `/Applications`. Are you sure there's nothing else to do here?

File: Casks/visual-studio.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+cask 'visual-studio' do
+  version 'Preview 1 (7.0 build 347),2016-11-16'
+  sha256 '7c674f7114e6a6532d6498dd8963cf1b5916badd694f251a11edf3024809bf48'
+
+  # xamarin.azureedge.net was verified as official when first introduced to the cask
+  url "https://xamarin.azureedge.net/VsMacInstaller/#{version.after_comma}/VisualStudioforMacPreviewInstaller.dmg"
+  name 'Visual Studio for Mac'
+  homepage 'https://www.visualstudio.com/vs/visual-studio-mac/'
+
+  installer manual: 'Install Visual Studio for Mac Preview.app/Contents/MacOS/Install_Xamarin'
```
Comment (jawshooah) [inline Casks/visual-studio.rb]: Just `installer manual: 'Install Visual Studio for Mac Preview.app'`

**Review by jawshooah — CHANGES_REQUESTED** [review]

File: Casks/visual-studio.rb (hunk 1 above)
Comment (joshka) [inline Casks/visual-studio.rb]: The installer downloads and installs a bunch of SDKs etc.
With that said, Visual Studio for Mac is Xamarin Studio rebadged which has the same problem, and it looks like some parts it installed are shared between the two making it not really possible to uninstall those parts sensibly on one or the other. What would you suggest here?

File: Casks/visual-studio.rb (hunk 2 above)
Comment (joshka) [inline Casks/visual-studio.rb]: Done

File: Casks/visual-studio.rb (hunk 1 above)
Comment (vitorgalvao) [inline Casks/visual-studio.rb]: @joshka Is there any reason why you’d have both? If not, just add the rest of the uninstall.

**Comment by adidalal** [comment]
Thank you for the contribution. It needed some fixes, so they were made in 721d04dc8304ff9215d8274797465b6fcd7baa89. Your contribution is still included (and still credited to you), with the appropriate modifications. Please feel free to ask about any of the changes.

File: Casks/visual-studio.rb (hunk 1 above)
Comment (vitorgalvao) [inline Casks/visual-studio.rb]: @adidalal Then we need to put said components in `zap`. As it is, `uninstall` is woefully incomplete, and there is no way to remove it correctly.

**Pull request closed without merging**